File: app/src/main/java/vergecurrency/vergewallet/helpers/utils/MathUtils.java
package vergecurrency.vergewallet.helpers.utils;

import java.io.EOFException;
import java.io.IOException;
import java.io.InputStream;
import java.math.BigDecimal;
import java.math.RoundingMode;

public final class MathUtils {


	public static double round(double value, int places) {
		if (places < 0) throw new IllegalArgumentException();

		BigDecimal bd = new BigDecimal(value);
		bd = bd.setScale(places, RoundingMode.HALF_UP);
		return bd.doubleValue();
	}

	public  static int getRandomNumber(int max) {
		return (int) (Math.random() * max);
	}

	public long readUnsignedInt(InputStream in) throws IOException {
		int ch1 = in.read();
		int ch2 = in.read();
		int ch3 = in.read();
		int ch4 = in.read();
		if ((ch1 | ch2 | ch3 | ch4) < 0) {
			throw new EOFException();
		}
		long ln4 = ch4 & 0x00000000ffffffffL;
		return (ln4 << 24) + (ch3 << 16) + (ch2 << 8) + (ch1 << 0);
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/helpers/utils/MathUtils.kt
package vergecurrency.vergewallet.helpers.utils

import java.io.EOFException
import java.io.IOException
import java.io.InputStream
import java.math.BigDecimal
import java.math.RoundingMode

object MathUtils {

    @Throws(IOException::class)
    fun readUnsignedInt(`in`: InputStream): Long {
        val ch1 = `in`.read()
        val ch2 = `in`.read()
        val ch3 = `in`.read()
        val ch4 = `in`.read()
        if (ch1 or ch2 or ch3 or ch4 < 0) {
            throw EOFException()
        }
        val ln4 = ch4 and 0x00000000ffffffffL.toInt()
        return (ln4 shl 24) + (ch3 shl 16).toLong() + (ch2 shl 8).toLong() + (ch1 shl 0).toLong()
    }


    fun round(value: Double, places: Int): Double {
        require(places >= 0)

        var bd = BigDecimal(value)
        bd = bd.setScale(places, RoundingMode.HALF_UP)
        return bd.toDouble()
    }

    fun getRandomNumber(max: Int): Int {
        return (Math.random() * max).toInt()
    }

}
